Assets/Scripts/SiteMarker.cs

```diff
@@ -26,6 +26,14 @@ public class SiteMarker : MonoBehaviour, IPointerEnterHandler, IPointerExitHandl
 
     }
 
+    private void OnTriggerEnter(Collider other) {
```

Comment:
Hopefully it's okay not to check that the other Collider is the one you're expecting. I don't know if you've setup layers so only the things you're expecting, but if so that's excellent.